tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Generators/ReadMeGeneratorTool.cs

```diff
@@ -0,0 +1,215 @@
+            ArgumentNullException.ThrowIfNull(serviceDocumentation);
+            ArgumentException.ThrowIfNullOrEmpty(outputPath);
+            ArgumentException.ThrowIfNullOrEmpty(engPath);
+
+            var readmeText = await File.ReadAllTextAsync(templatePath);
+
+            var messages = new ChatMessage[]{
+                new SystemChatMessage(
+                    string.Join("\n",
+                        "We're going to create some READMEs.",
+                        "The parameters are:",
+                        "* A URL that contains service documentation, to be used when creating key concepts, an introduction blurb and any other places where conceptual docs are needed",
+                        "* A package path that we can use to generate documentation links",
+                        "Here are some more rules to follow:",
+                        "- Do not touch the following sections, or its subsections: Contributing.",
+                        "- Do not generate sample code.",
+                        "- Rules for proper readmes can be found here: https://github.com/Azure/azure-sdk/blob/main/docs/policies/README-TEMPLATE.md"
+                    )
+                ),
+                new SystemChatMessage("The readme template is this: \"" + readmeText + "\" which we fill in with the user parameters, which follow:"),
+                new UserChatMessage($"Service URL: {serviceDocumentation}"),
+                new UserChatMessage($"Package path: {packagePath}")
+            };
+
+            var response = await chatClient.CompleteChatAsync(messages);
+            var generatedReadmeText = Customize(response.Value.Content[0].Text);
+
+            await File.WriteAllTextAsync(outputPath, generatedReadmeText);
+            await Validate(engPath: engPath, readmePath: outputPath);
+
+            output.Output($"Readme written to {outputPath}");
+        }
+
+        private async Task Validate(string engPath, string readmePath)
+        {
+            output.Output($"Running {engPath}/common/scripts/Verify-Links.ps1 \"{readmePath}\"");
+
+            // TODO: I think chradek has a replacement for this.
+            var process = Process.Start(new ProcessStartInfo()
+            {
+                FileName = Environment.OSVersion.Platform == PlatformID.Win32NT ? "powershell.exe" : "pwsh",
+                Arguments = $"\"{Path.Join(engPath, "common", "scripts", "Verify-Links.ps1")}\" \"{readmePath}\"",
+                RedirectStandardError = true,
+                RedirectStandardOutput = true
+            });
+
+            ArgumentNullException.ThrowIfNull(process);
+
+            await process.WaitForExitAsync();
+
+            if (process.ExitCode != 0)
+            {
+                //var stderr = await process.StandardError.ReadToEndAsync();
+                var stdout = await process.StandardOutput.ReadToEndAsync();
+                var tabified = string.Join("\n  ", stdout.Split("\n"));
```

Comment:
```suggestion
                var tabified = stdout.Replace("\n", "\n  ");
```

Split/join is going to create a ton of unnecessary allocations.  Doing an inline replace will be much more efficient here.

Reply:
I feel ashamed on this one.